client/src/registry.rs

```diff
@@ -15,6 +15,8 @@ pub struct NodeEntry {
     /// To change Session, we need to replace whole NodeEntry.
     pub session: Arc<Session>,
     pub slot: SlotId,
+    /// Connection lock
+    pub conn_lock: Arc<RwLock<()>>,
```

Comment:
This should be in `VirtualNode`, because it's guards only TCP connections

Reply:
This lock needs to be shared. We create a new `VirtualNode` each time before we connect